jablib/src/test/java/org/jabref/logic/util/io/FileUtilTest.java

```diff
@@ -236,6 +236,25 @@ void getFileNameWithMultipleDotsString() {
         assertEquals("te.st", FileUtil.getBaseName("te.st.PdF  "));
     }
 
+    @Test
+    void getFileNameFromUrlsCorrectly() {
+        final String urls[] = {"www.example.com/", "http://www.example.com/", "https://www.example.com/"};
+        final String dirs[] = {"path/to/", "not\\a\\windows\\path/", "///", ""};
+        final String files[] = {"file.pdf", "blank", "unknown.doc", ""};
+        final String queries[] = {"", "?field=value", "?a=1&b=2", "?search=for+a+file"};
+        for (String file : files) {
+            for (String dir : dirs) {
+                final String path = dir + file;
+                for (String url : urls) {
+                    final String webpage = url + path;
+                    for (String query : queries) {
+                        assertEquals(file, FileUtil.getFileNameFromUrl(webpage + query), "from '" + webpage + query + "'");
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Please convert to `@ParameterizedTest` and `@MethodSource`. You can convert a nice `Stream` of `Argument` there. Nesting inside a test itself is not current SOTA.